app/src/main/java/com/yacgroup/yacguide/TourbookActivity.kt

```diff
@@ -132,17 +118,70 @@ class TourbookActivity : BaseNavigationActivity() {
         }
     }
 
-    private fun _import() {
-        _ioOption = IOOption.eImport
-        if (FilesystemUtils.checkPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)) {
-            _showFileChooser()
+    @Throws(IOException::class)
+    private fun _readTextFromUri(uri: Uri): String {
+        val stringBuilder = StringBuilder()
+        contentResolver.openInputStream(uri)?.use { inputStream ->
+            BufferedReader(InputStreamReader(inputStream)).use { reader ->
+                var line: String? = reader.readLine()
+                while (line != null) {
+                    stringBuilder.append(line)
+                    line = reader.readLine()
+                }
+            }
+        }
+        return stringBuilder.toString()
+    }
+
+    private fun _writeTextToUri(uri: Uri, text: String) {
+        try {
+            contentResolver.openFileDescriptor(uri, "w")?.use {
+                FileOutputStream(it.fileDescriptor).use {
+                    it.write(text.toByteArray())
+                }
+            }
+        } catch (e: FileNotFoundException) {
+            e.printStackTrace()
+        } catch (e: IOException) {
+            e.printStackTrace()
+        }
+    }
+
+    private fun _import(uri: Uri) {
+        val confirmDialog = Dialog(this)
+        confirmDialog.setContentView(R.layout.dialog)
+        confirmDialog.findViewById<TextView>(R.id.dialogText).text = getString(
+                R.string.override_tourbook)
+        confirmDialog.findViewById<View>(R.id.yesButton).setOnClickListener {
+            try {
+                val jsonString = _readTextFromUri(uri)
+                TourbookExporter(_db).importTourbook(jsonString)
+                Toast.makeText(this, getString(R.string.tourbook_import_successfull),
+                        Toast.LENGTH_SHORT).show()
+            } catch (e: JSONException) {
```

Comment:
You should also catch the `IOExcetion` that might be thrown by `_readTextFromUri()`